entity-management-web/src/main/java/eu/europeana/entitymanagement/web/EMController.java

```diff
@@ -721,8 +723,31 @@ private ResponseEntity<String> createResponseRetrieve(
       String languages,
       String contentType)
       throws EuropeanaApiException {
-    EntityRecord entityRecord =
-        entityRecordService.retrieveEntityRecord(type, identifier.toLowerCase(), true);
+    
+    EntityRecord entityRecord = null;
+    try {
+      entityRecord = 
+          entityRecordService.retrieveEntityRecord(type, identifier.toLowerCase(), true);
+    }
+    catch (EntityNotFoundException ex) {
+      //search in the coreferences of all enabled entities
+      String entityUri = EntityRecordUtils.buildEntityIdUri(type, identifier);
+      List<EntityRecord> corefEntities = entityRecordService.findAllEntitiesDupplicationByCoreference(Collections.singletonList(entityUri));
+      if(corefEntities.size()>1) {
+        String allCorefUris = corefEntities.stream().map(el -> el.getEntityId()).collect(Collectors.joining(","));
+        throw new MultipleChoicesException(String.format(MULTIPLE_CHOICES_FOR_REDIRECTION_MSG, entityUri, allCorefUris));
+      }
+      else if(corefEntities.size()==1) {
+        //return 301 redirect
+        return ResponseEntity.status(HttpStatus.MOVED_PERMANENTLY)
+            .location(URI.create(corefEntities.get(0).getEntityId()))
+            .build();
+      }
+      else {
+        throw new EntityNotFoundException(entityUri);
+      }
+    }
+    
     if (entityRecord.isDisabled()) {
       return redirectDeprecated(entityRecord, type, identifier.toLowerCase());
```

Comment:
this method should be refactored as well within this ticket, The service should implement the business logic to select the redirection URI and the controller should only build the response entity